x-pack/plugins/observability/server/services/slo/transform_generators/common.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import { InvalidTransformError } from '../../../errors';
+
+export function getElastichsearchQueryOrThrow(kuery: string) {
+  try {
+    return toElasticsearchQuery(fromKueryExpression(kuery));
+  } catch (err) {
+    throw new InvalidTransformError(`Invalid KQL: ${kuery}`);
```

Comment:
Shall we also return `err.message` as part of the error? I think it will be hard to debug without knowing why kuery is invalid.

Reply:
https://github.com/elastic/kibana/issues/151297